fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/block/type/BlockSetTypeBuilder.java

```diff
@@ -0,0 +1,238 @@
+	 *
+	 * <p>Defaults to {@link SoundEvents#BLOCK_WOODEN_BUTTON_CLICK_OFF}.
+	 *
+	 * @return this builder for chaining
+	 */
+	public BlockSetTypeBuilder buttonClickOffSound(SoundEvent buttonClickOffSound) {
+		this.buttonClickOffSound = buttonClickOffSound;
+		return this;
+	}
+
+	/**
+	 * Sets this block set type's button click on sound.
+	 *
+	 * <p>Defaults to {@link SoundEvents#BLOCK_WOODEN_BUTTON_CLICK_ON}.
+	 *
+	 * @return this builder for chaining
+	 */
+	public BlockSetTypeBuilder buttonClickOnSound(SoundEvent buttonClickOnSound) {
+		this.buttonClickOnSound = buttonClickOnSound;
+		return this;
+	}
+
+	/**
+	 * Creates a new {@link BlockSetTypeBuilder} that copies all of another builder's values.
+	 *
+	 * @param builder the {@link BlockSetTypeBuilder} whose values are to be copied
+	 *
+	 * @return the created copy
+	 */
+	public static BlockSetTypeBuilder copyOf(BlockSetTypeBuilder builder) {
+		BlockSetTypeBuilder copy = new BlockSetTypeBuilder();
+		copy.openableByHand(builder.openableByHand);
+		copy.soundGroup(builder.soundGroup);
+		copy.doorCloseSound(builder.doorCloseSound);
+		copy.doorOpenSound(builder.doorOpenSound);
+		copy.trapdoorCloseSound(builder.trapdoorCloseSound);
+		copy.trapdoorOpenSound(builder.trapdoorOpenSound);
+		copy.pressurePlateClickOffSound(builder.pressurePlateClickOffSound);
+		copy.pressurePlateClickOnSound(builder.pressurePlateClickOnSound);
+		copy.buttonClickOffSound(builder.buttonClickOffSound);
+		copy.buttonClickOnSound(builder.buttonClickOnSound);
+		return copy;
+	}
+
+	/**
+	 * Creates a new {@link BlockSetTypeBuilder} that copies all of another block set type's values.
+	 *
+	 * @param setType the {@link BlockSetType} whose values are to be copied
+	 *
+	 * @return the created copy
+	 */
+	public static BlockSetTypeBuilder copyOf(BlockSetType setType) {
+		BlockSetTypeBuilder copy = new BlockSetTypeBuilder();
+		copy.openableByHand(setType.canOpenByHand());
+		copy.soundGroup(setType.soundType());
+		copy.doorCloseSound(setType.doorClose());
+		copy.doorOpenSound(setType.doorOpen());
+		copy.trapdoorCloseSound(setType.trapdoorClose());
+		copy.trapdoorOpenSound(setType.trapdoorOpen());
+		copy.pressurePlateClickOffSound(setType.pressurePlateClickOff());
+		copy.pressurePlateClickOnSound(setType.pressurePlateClickOn());
+		copy.buttonClickOffSound(setType.buttonClickOff());
+		copy.buttonClickOnSound(setType.buttonClickOn());
+		return copy;
+	}
+
+	/**
+	 * Builds and registers a {@link BlockSetType} from this builder's values.
+	 *
+	 * @param id the id for the built {@link BlockSetType}
+	 *
+	 * @return the built and registered {@link BlockSetType}
+	 */
+	public BlockSetType build(Identifier id) {
+		return BlockSetType.register(new BlockSetType(id.toString(), openableByHand, soundGroup, doorCloseSound, doorOpenSound, trapdoorCloseSound, trapdoorOpenSound, pressurePlateClickOffSound, pressurePlateClickOnSound, buttonClickOffSound, buttonClickOnSound));
```

Comment:
Would be nice to have a way to build it without registering it, maybe?

Reply:
What purpose would that serve? I only forsee that causing problems where people forget to register it and it causes issues.

Reply:
If anything, a method named `build` should not `register`

Reply:
I just thought the name `buildAndRegister` was too verbose. There's no sense having an unregistered type, and being able to build without registering I feel would lead to too many people forgetting to register.
Keep in mind it's not an actual Registry, it's just adding to the internal `private static final Set<BlockSetType> VALUES` (or the WoodType equivalent).
I suppose I could remove the call to `register` and just return the built type, and rely on modders to call `BlockSetType.register` themselves... would that be preferred?

Reply:
Or the method could be called `register`? IMO it's clear from the return type that it's built.

Reply:
Yeah I suppose that works too. I guess I just went "it's a builder -> it ends with `build`" but `register` would be a fine name. Is that good with you @Technici4n ?

Reply:
could be `buildAndRegister` if it does both. :P

Reply:
or just `register` sure

Reply:
Decided to make `build` and `register` two different methods - `register` calling `build` and registering its returned type, and `build` simply making the type without registering. Added javadoc to (hopefully!) make this clear.